src/job.py

```diff
@@ -74,12 +77,29 @@ async def run(self) -> None:
             No exception is raised for empty result sets, only a warning is logged.
 
         """
+        run_id = uuid.uuid4().hex
+
+        start = perf_counter()
+
         df = await self.source.fetch()
         if not self.source.is_empty(df):
             self.destination.save(df)
         else:
+            # TODO should this case still submit metrics?
```

Comment:
I think its ok to record the run since it might have taken a while to determine if the result set was empty.

Might also be cute to have a counter on jobs with no results (it might be an alert worthy case if there are recurring instances of no results).